x-pack/plugins/security_solution/public/detections/pages/detection_engine/detection_engine.tsx

```diff
@@ -168,7 +184,31 @@ const DetectionEnginePageComponent: React.FC<DetectionEngineComponentProps> = ({
     sourcererDataView,
     runtimeMappings,
     loading: isLoadingIndexPattern,
+    indexPattern,
   } = useSourcererDataView(SourcererScopeName.detections);
+  const spaceId = useSpaceId();
+
+  const [filterControlsUrlState, setFilterControlsUrlState] = useState<FilterControlConfig[]>();
+
+  const onUrlParamInit = (param: FilterControlConfig[] | null) => {
```

Comment:
I guess this new way makes sense. But there should be a way to sync URL back when user update the controls manually?

Something that replaces `useFilterUpdatesToUrlSync`

Reply:
I guess the main component, could take a `boolean` prop called `shouldSyncToUrl` and a `url` param key which can handle all this.

Alternatively, instead of writing all this custom logic, you can simple make use of `createKbnUrlStateStorage` which sync from and to URL automatically.

https://github.com/elastic/kibana/blob/24de5e499c7bdc1239b0fec5c0a2a32a65c93e42/src/plugins/kibana_utils/public/state_sync/state_sync_state_storage/create_kbn_url_state_storage.ts#L53-L73


Since URL sync is only done conditionaly, we even have a implementation which can be probably used as a reference

https://github.com/elastic/kibana/blob/24de5e499c7bdc1239b0fec5c0a2a32a65c93e42/packages/kbn-expandable-flyout/src/provider.tsx#L25-L36

Reply:
Since different solutions have different ways of persisting data in URLs, I handled this outside the `AlertFilterControls` component, in particular in [DetectionEngineFilters](https://github.com/umbopepato/kibana/blob/3c3e8e13fc3b79e7cbf7ebad5a425e1a42581be2/x-pack/plugins/security_solution/public/detections/components/detection_engine_filters/detection_engine_filters.tsx). Two way sync should work correctly now

Reply:
That is perfect. Thanks for implementing it.